chromium_src/chrome/browser/chrome_browser_main.cc

```diff
@@ -6,8 +6,143 @@
+      std::string value =
+          command_line->GetSwitchValueASCII(switches::kSetUserColor);
+      std::vector<std::string> rgb = base::SplitString(
+          value, ",", base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+      if (rgb.size() == 3) {
+        int r, g, b;
+        if (base::StringToInt(rgb[0], &r) && base::StringToInt(rgb[1], &g) &&
+            base::StringToInt(rgb[2], &b) && r >= 0 && r <= 255 && g >= 0 &&
+            g <= 255 && b >= 0 && b <= 255) {
+          theme_service->SetUserColor(SkColorSetRGB(r, g, b));
+        }
+      }
+    }
+
+    if (command_line->HasSwitch(switches::kSetColorScheme)) {
+      std::string scheme =
+          command_line->GetSwitchValueASCII(switches::kSetColorScheme);
+      static constexpr auto kSchemeMap =
+          base::MakeFixedFlatMap<std::string_view,
+                                 ThemeService::BrowserColorScheme>({
+              {"system", ThemeService::BrowserColorScheme::kSystem},
+              {"light", ThemeService::BrowserColorScheme::kLight},
+              {"dark", ThemeService::BrowserColorScheme::kDark},
+          });
+
+      auto it = kSchemeMap.find(scheme);
+      if (it != kSchemeMap.end()) {
+        theme_service->SetBrowserColorScheme(it->second);
+      }
+    }
+
+    if (command_line->HasSwitch(switches::kSetGrayscaleTheme)) {
+      std::string value =
+          command_line->GetSwitchValueASCII(switches::kSetGrayscaleTheme);
+      bool is_grayscale = (value == "true");
+      theme_service->SetIsGrayscale(is_grayscale);
+    }
+
+    if (command_line->HasSwitch(switches::kSetColorVariant)) {
+      std::string variant =
+          command_line->GetSwitchValueASCII(switches::kSetColorVariant);
+      static constexpr auto kVariantMap =
+          base::MakeFixedFlatMap<std::string_view,
+                                 ui::mojom::BrowserColorVariant>({
+              {"tonal_spot", ui::mojom::BrowserColorVariant::kTonalSpot},
+              {"neutral", ui::mojom::BrowserColorVariant::kNeutral},
+              {"vibrant", ui::mojom::BrowserColorVariant::kVibrant},
+              {"expressive", ui::mojom::BrowserColorVariant::kExpressive},
+          });
+
+      auto it = kVariantMap.find(variant);
+      if (it != kVariantMap.end()) {
+        theme_service->SetBrowserColorVariant(it->second);
+      }
+    }
+#endif
+  }
+
+  static void ProcessForProfile(const base::CommandLine* command_line,
+                                Profile* profile) {
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_CHROMEOS)
+    if (!command_line || !profile) {
+      return;
+    }
+
+    ThemeService* theme_service = ThemeServiceFactory::GetForProfile(profile);
+    if (!theme_service) {
+      return;
+    }
+
+    ProcessThemeCommandLineSwitches(command_line, theme_service);
+#endif
+  }
+
+  ThemeCommandLineHandler() = delete;
+  ~ThemeCommandLineHandler() = delete;
+};
```

Comment:
Since this class no members, and the functions being called are all static, we should make these functions outside the class to the namespace level:

https://google.github.io/styleguide/cppguide.html#Nonmember,_Static_Member,_and_Global_Functions

> Sometimes it is useful to define a function not bound to a class instance. Such a function can be either a static member or a nonmember function. Nonmember functions should not depend on external variables, and should nearly always exist in a namespace. Do not create classes only to group static members; this is no different than just giving the names a common prefix, and such grouping is usually unnecessary anyway.